extensions/elasticsearch/src/main/scala/elasticsearchExtensions.scala

```diff
@@ -27,36 +31,68 @@ trait EsStream {
                    (implicit es: EsClient, ec: ExecutionContextForBlockingOps): Source[String, NotUsed] = {
+    queryAsStream(query, index, `type`, scrollKeepAlive, scrollSize, Array.empty, Array.empty, List())
+  }
+  /**
+   * Get the result of a search query as a stream.
+   * Internally, the scan/scroll ES API is used to improve performance when fetching large chunks of data.
+   * @param query search query
+   * @param index ES index
+   * @param `type` ES type
+   * @param scrollKeepAlive ES scroll keep alive
+   * @param scrollSize ES scroll size
+   * @param included
+   * @param excluded
+   * @param addField
+   * @param es
+   * @param ec
+   * @return
+   */
+  def queryAsStream(query: QueryBuilder,
+                    index: String,
+                    `type`: String,
+                    scrollKeepAlive: FiniteDuration,
+                    scrollSize: Int,
+                    included: Array[String],
+                    excluded: Array[String],
+                    addField: List[String])
+                   (implicit es: EsClient, ec: ExecutionContextForBlockingOps): Source[String, NotUsed] = {
     implicit val ecValue = ec.value
 
-    lazy val futInitSearchResp = Future {
-      es
+    def req(query: QueryBuilder) = {
+      val srb = es
         .prepareSearch(index)
         .setTypes(`type`)
-        .setSearchType(SearchType.SCAN)
-        .setScroll(scrollKeepAlive.toMillis.toString)
+        .setQuery(query)                 // Query
+        .setScroll(new TimeValue(scrollKeepAlive.toMillis))
         .setSize(scrollSize)
-        .setQuery(query)
-        .get()
+
+      addField.foreach { x => srb.addField(x) }
+      if (!included.isEmpty || !excluded.isEmpty) srb.setFetchSource(included, excluded)
+
+      val res = srb
+        .execute()
+        .actionGet()
+
+      val docs = res.getHits.getHits.toVector.map(_.getSourceAsString)
+
+      docs #:: next(res)
     }
+ 
+    def next(res: SearchResponse): Stream[Vector[String]] = {
+      val newRes = es
+        .prepareSearchScroll(res.getScrollId())
+        .setScroll(new TimeValue(scrollKeepAlive.toMillis))
+        .execute()
+        .actionGet
+
+      val docs = newRes.getHits.getHits.toVector.map(_.getSourceAsString)
 
-    SourceExt.seededLazyAsync(futInitSearchResp) { initSearchResp =>
-      SourceExt.unfoldPullerAsync(initSearchResp) { searchResp =>
-        Future {
-          es
-            .prepareSearchScroll(searchResp.getScrollId)
-            .setScroll(scrollKeepAlive.toMillis.toString)
-            .get
-        }
-        .map { nextSearchResp =>
-          val docs = nextSearchResp.getHits.getHits.toVector.map(_.getSourceAsString)
-
-          if (docs.isEmpty) None -> None
-          else Some(docs) -> Some(nextSearchResp)
-        }
-      }
+      if (docs.isEmpty) Stream.Empty
+      else docs #:: next(newRes)
     }
-    .mapConcat(identity)
+
+    Source(req(query)).mapConcat { identity }
```

Comment:
Initial response started immediatelly. If Source declared via queryAsStream much earlier (scrollKeepAlive) than Materialization (runWith(Sink)), than cursor may outdate.

Correct code is like this `Source.fromIterator(() => req(query).toIterator).mapConcat(identity)`